k2/python/k2/fsa.py

```diff
@@ -1061,19 +1062,17 @@ def shape(self) -> Tuple[int, ...]:
             raise ValueError(f'Unsupported num_axes: {self.arcs.num_axes()}')
 
     @classmethod
-    def from_str(cls, s: str) -> 'Fsa':
+    def from_str(cls, s: str, num_aux_labels: int = 0,
+                 aux_label_names: List[str] = ['aux_labels',
```

Comment:
Are there any reasons to set these kinds of default names? 
`Optional[List[str]]` seems to be more natural.

`num_aux_labels` and `len(aux_label_names)` contain the
same information, though it is not checked in the code.

Reply:
mm, I see what you mean...  I'll make it so at most one should be specified.